src/main/java/net/fabricmc/loom/util/Platform.java

```diff
@@ -24,14 +24,36 @@
 
 package net.fabricmc.loom.util;
 
-public record Architecture(String name) {
-	public static final Architecture CURRENT = new Architecture(System.getProperty("os.arch"));
+public interface Platform {
```

Comment:
What's the point of making this an interface with a single impl? Mocking?

Reply:
Yes, Mocking. Previously it was impossible to test any of this as it changed based upon the platform its running on. I will be able to write tests for the hardware that I dont use regularly.